footbot/optimiser/team_selector.py

```diff
@@ -215,6 +215,9 @@ def select_team(
     squad_prob = cp.Problem(cp.Maximize(objective), constraints)
     squad_prob.solve(solver="GLPK_MI")
 
+    if squad_prob.status != "optimal":
+        raise Exception("Unable to find optimal team selection")
```

Comment:
lets add

```
class NoOptimalTeamSelection(Exception):
   ...
```

so we can specifically catch this error instead of any unexpected ones